python/ray/experimental/channel/torch_tensor_nccl_channel.py

```diff
@@ -49,21 +50,18 @@ def __init__(
         self._writer_registered: bool = False
         self._reader_registered: bool = False
 
-        self._device = device
-        if self._device is None:
-            # TODO(swang): Allow default device to be overridden.
-            self._device = torch_utils.get_devices()[0]
-        if self._device.type != "cuda":
-            raise ValueError(
-                f'Actor\'s default device has type "{self._device.type}", need "cuda"'
-            )
+        # TODO(swang): Allow default device to be overridden.
+        self._device = _get_default_torch_device()
```

Comment:
Isn't this supposed to be self._typ.device?

Reply:
Ah TorchTensorType doesn't have a device right now. I think that would be too annoying to specify, you'd have to know which device the actor was using.